docs/_source/docs/architecture.rst

```diff
@@ -135,15 +135,222 @@ to demonstrate ways of organising different projects.
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: yaml
+
+    # config/config.yaml
+
+    # This is the top-level config file for the whole project. Every stack in the project will
+    # inherit these settings unless they explicitly override them. At minimum, you need the
+    # project_code and region, but you can set whatever StackGroup and Stack configurations you
+    # want inherited by the rest of the project.
+
+    # By making the project-code parameterized by environment name, it guarantees that one
+    # environment's stacks do not conflict with another environment's stacks.
+    project_code: indigo-{{ var.environment }}
+    region: {{ var.region }}
+
+    # Setting the template_bucket_name with a !stack_output means that every stack in the project
+    # (except the deployment stack) will have its template uploaded to that bucket. It ALSO means
+    # that every stack in the project will inherit a dependency on the deployment stack (except the
+    # deployment stack... see its comments below)
+    template_bucket_name: !stack_output project/deployment.yaml::BucketName
+
+    # By making the template_key_prefix parameterized by environment, it means that every
+    # environment's templates will be uploaded to its own directory structure within the template
+    # bucket, preventing conflicts.
+    template_key_prefix: {{ var.environment }}
+
+    # Setting the iam_role on the project like this means that all Sceptre actions on all stacks in
+    # the project will be done by assuming the deployment role whose ARN is output by the deployment
+    # stack (except for actions on the deployment stack... see its comments below).
+    iam_role: !stack_output project/deployment.yaml::DeploymentRoleArn
+
+
+The Deployment Stack
+^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: yaml
+
+    # config/project/deployment.yaml
+
+    # This is a StackConfig for a stack that defines the project's basic requirements, that all
+    # stacks and all environments will depend on for deployment to work. Because the project config
+    # references this stack using !stack_output, all OTHER stacks in the project will depend upon
+    # this one.
+
+    # We override the project_code here so that there will only be ONE stack for deployment that is
+    # shared across all environments.
+    project_code: indigo-deployment
+
+    # Setting is_project_dependency has a few important effects for this stack
+    #   1. It disables dependencies. In other words, this stack cannot depend on any other stack.
+    #       This is how it bypasses the circular dependency that it would otherwise inherit from
+    #       the project's config (config/config.yaml)
+    #   2. It disables the !stack_output resolver, making it resolve to nothing. In other words, it
+    #       cannot inherit the template_bucket_name or iam_role from the project config. This is
+    #       good because this stack DEFINES the template bucket and iam_role.
+    is_project_dependency: True
```

Comment:
having to define this key feels kinda clunky to me.  I'm wondering if this should actually be a template type instead?   Something like a `project` or `base` type so we can identify it with something like this..

```
template:
  type: project    # or a "base" type?
  path: project/deployment.yaml
``` 

Another option is to identify it with a special filename, like `project.yaml` or `base.yaml`?

Reply:
> having to define this key feels kinda clunky to me.

I've only defined it here in the docs for explanatory purposes, as I've explained _all_ the things. This config has been documented elsewhere on the Stack Config docs. Could you elaborate on what you mean by this?

Regarding a special template type, I don't think that option would be viable; the distinction isn't in the _template_, it's in the relation of this stack to the other stacks. Also, `is_project_dependency` doesn't do anything with the stack template. Instead, it alters the way dependencies are handled and the way the !stack_output resolver operates. 

Regarding the "special filename" approach you suggest, that seems to indicate a completely different class of stack. It also would also likely prohibit multiple stacks that are indicated as project dependencies. I'm not crazy about the idea of the config reader needing to check the stack name for every stack, and if it's "project" or "base", instantiating a different kind of stack or something like that.

Rather than any of those, perhaps we could rename the configuration to `disable_dependencies`. Would you find that more self-explanatory?

Reply:
I agree that `is_project_dependency` does feel clunky as it's a way of marking a special template. But I can't think of a better way to handle this at the moment.

As for the naming I think having a name that self explains what it does would be more useful as `is_project_dependency` makes me immediately have to look for docs. `disable_dependencies` is a bit more clear for sure or maybe `disable_stack_dependencies` or `is_root_template` or something along those lines?